lightning/src/ln/outbound_payment.rs

```diff
@@ -961,7 +961,16 @@ impl OutboundPayments {
 					let payment_hash = PaymentHash(Sha256::hash(&keysend_preimage.0).to_byte_array());
 					let payment_release_secret = entropy_source.get_secure_random_bytes();
 					let pay_params = PaymentParameters::from_static_invoice(invoice);
-					let route_params = RouteParameters::from_payment_params_and_value(pay_params, amount_msat);
+					let mut route_params = RouteParameters::from_payment_params_and_value(pay_params, amount_msat);
+
+					if let Err(()) = onion_utils::set_max_path_length(
+						&mut route_params, &RecipientOnionFields::spontaneous_empty(), Some(keysend_preimage),
+						best_block_height
+					) {
+						abandon_with_entry!(entry, PaymentFailureReason::UnexpectedError);
+						return Err(Bolt12PaymentError::SendingFailed(RetryableSendFailure::OnionPacketSizeExceeded))
+					}
+
```

Comment:
Do we plan to call `find_initial_route` later? Currently, that calls `set_max_path_length`.

Reply:
Moved to calling `find_initial_route`, but I think we need to call `set_max_path_length` here because per the commit message it would be good to error on payment initiation and not wait until the recipient comes back online. IMO it's not too bad to call it redundantly but open to input.